datamade/erikson-edi
# Dockerize

## Overview

Sets up docker configuration, updates settings files accordingly

* [x] Adds Dockerfile, docker-compose.yml files
* [x] Updates `settings_local.example.py` file
* [x] Updates Readme with setup instructions
* [x] All import commands run successfully
* [x] Updates `requirements.txt` dependencies to appease dependabot (#147 and #128)

Closes #154 

### Notes

I can get everything to work except the `import_geographies` command when trying to execute the command line `shp2pgsql` line. I believe I've included all the necessary dependencies for it, but it is exiting with status 2 - meaning it can not parse one or more of the command line arguments:

```
Ephesus:erikson-edi derekeder$ docker-compose run --rm app ./manage.py import_geographies
Starting edi-postgres ... done
/usr/local/lib/python3.7/site-packages/sqlalchemy/dialects/postgresql/base.py:2399: SADeprecationWarning: The create_engine.convert_unicode parameter and corresponding dialect-level parameters are deprecated, and will be removed in a future release.  Modern DBAPIs support Python Unicode natively and this parameter is unnecessary.
Field aland is an FTDouble with width 14 and precision 0
Field awater is an FTDouble with width 14 and precision 0
Shapefile type: Polygon
Postgis type: MULTIPOLYGON[2]
psql: could not connect to server: No such file or directory
	Is the server running locally and accepting
	connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?
Traceback (most recent call last):
  File "/app/edi/management/commands/import_geographies.py", line 74, in handle
    subprocess.run(cmd, check=True, universal_newlines=True, shell=True)
  File "/usr/local/lib/python3.7/subprocess.py", line 512, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command 'shp2pgsql -c -s 4296 data/final/shapefiles/cb_2016_17_tract_500k.shp illinois_geography | psql -U postgres -d edi' returned non-zero exit status 2.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./manage.py", line 15, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.7/site-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/base.py", line 323, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/base.py", line 364, in execute
    output = self.handle(*args, **options)
  File "/app/edi/management/commands/import_geographies.py", line 76, in handle
    raise subprocess.CalledProcessError(str(e))
TypeError: __init__() missing 1 required positional argument: 'cmd'
```

## Testing Instructions

* Go through the Readme and ensure that all steps complete without error
* Run all import commands
* Test that the site works locally with initial data on `localhost:8000`
